integration/test_database_operations.py

```diff
@@ -0,0 +1,63 @@
+"""Database operations tests."""
```

Comment:
Does this test need a running server? If not please move to client/tests, i.e. convert to module test instead of integration.

Reply:
No, it doesn't need server. I can rewrite it in a more unit test vibe.

Reply:
Or is moving it to the client enough?